datastructures/src/main/java/com/aparamonov/algostructure/list/ListUsage.java

```diff
@@ -0,0 +1,145 @@
+        test.removeFromList(linkedList);
+
+    }
+
+    /**
+     * # Generates file and fills it with UUID per line
+     *
+     * @param linesAmount amount of UUID lines to generate
+     * @param filePath path to file on filesystem
+     */
+    public void generateFileWithData(int linesAmount, String filePath) {
+        try {
+            Path path = Paths.get(filePath);
+            Files.deleteIfExists(path);
+            try (BufferedWriter writer = Files.newBufferedWriter(path)) {
+                for(int i = 0; i < linesAmount; i++) {
+                    String uuid = UUID.randomUUID().toString();
+                    writer.write(uuid);
+                    writer.newLine();
+                }
+                System.out.println("Finished write data to file");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * # Appends lines from given file to the END of the list
+     *
+     * @param filePath path to file which contains data
+     * @param list datastructure - implementation of List interface:
+     *             - ArrayList
+     *             - LinkedList
+     */
+    public void appendToEnd(String filePath, List<String> list) {
+        System.out.println("Appending to the end...");
+
+        long startTime = System.currentTimeMillis();
+        try (Stream<String> stream = Files.lines(Paths.get(filePath))) {
+            stream.forEach(list::add);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        long endTime = System.currentTimeMillis();
+
+        System.out.println("Append to the end took: " + measureTime(startTime, endTime));
+    }
+
+    /**
+     * # Appends lines from given file to the MIDDLE of the list
+     *
+     * @param filePath path to file which contains data
+     * @param list datastructure - implementation of List interface:
+     *             - ArrayList
+     *             - LinkedList
+     */
+    public void appendToMiddle(String filePath, List<String> list) {
+        System.out.println("Appending to the middle...");
+        System.out.println("List size: " + list.size());
+
+        long startTime = System.currentTimeMillis();
+        try (Stream<String> stream = Files.lines(Paths.get(filePath))) {
+            stream.forEach(l -> list.add(list.size()/2, l));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        long endTime = System.currentTimeMillis();
+
+        System.out.println("Append to the middle took: " + measureTime(startTime, endTime));
+    }
+
+    /**
+     * # Removes lines from given list
+     *
+     * @param list datastructure - implementation of List interface:
+     *             - ArrayList
+     *             - LinkedList
+     */
+    public void removeFromList(List<String> list) {
+        System.out.println("Removing lines from the list...");
+
+        long startTime = System.currentTimeMillis();
```

Comment:
one more thing to add here - consider using JMH for microbenchmarking of java.util classes. first of all, this tool has stable knowledge and code base, secondly - it's commonly used in production